personal/privacyidea/default/mfa_intro.tpl

```diff
@@ -104,6 +104,26 @@
 
 {if ($parent != "roletabs")}
 {if $manageTokensACL}
+<script>
+var changesDoneNotificationArmed = false;
+function armChangesDoneNotif()
+{
+    if (changesDoneNotificationArmed === false) {
+        document.forms.mainform.addEventListener("submit", (e) => {
+            if (e.submitter.name === "add_token") {
+                let result = confirm("{t}You have unsaved changes, are you sure you want to continue? All unsaved changes to this account will be lost.{/t}");
+                if (!result) {
+                    e.preventDefault();
+                    return;
+                }
+            }
+        });
+    }
+
+    changesDoneNotificationArmed = true;
+}
```

Comment:
Please use the following and delete the `onclick` handlers:

```suggestion
document.forms.mainform.addEventListener("submit", (e) => {
    if (e.submitter.name === "add_token") {
        for (let el of document.forms.mainform.querySelectorAll("input[type='checkbox']")) {
            if (el.checked != (el.getAttribute("checked") !== null)) {
                if (!confirm("{t}You have unsaved changes, are you sure you want to continue? All unsaved changes to this account will be lost.{/t}")) {
                    e.preventDefault();
                    break;
                }
            }
        }
    }
});
```

Reply:
@dzatoah please follow-up